cloud/scope/powervs_cluster.go

```diff
@@ -1084,114 +1124,146 @@ func (s *PowerVSClusterScope) createVPCSubnet(subnet infrav1beta2.Subnet) (*stri
 		tg, _, err := s.TransitGatewayClient.GetTransitGateway(&tgapiv1.GetTransitGatewayOptions{
 			ID: s.GetTransitGatewayID(),
 		})
 		if err != nil {
-			return err
+			return false, err
 		}
-		err = s.checkTransitGatewayStatus(tg.ID)
+		requeue, err := s.checkTransitGatewayStatus(tg.ID)
 		if err != nil {
-			return err
+			return false, err
 		}
-		return nil
+		return requeue, nil
 	}
 
 	// check transit gateway exist in cloud
-	tgID, err := s.checkTransitGateway()
+	tgID, requeue, err := s.checkTransitGateway()
 	if err != nil {
-		return err
+		return false, err
 	}
 	if tgID != "" {
 		s.SetStatus(infrav1beta2.ResourceTypeTransitGateway, infrav1beta2.ResourceReference{ID: &tgID, ControllerCreated: ptr.To(false)})
-		return nil
+		return requeue, nil
 	}
 	// create transit gateway
 	transitGatewayID, err := s.createTransitGateway()
 	if err != nil {
-		return err
+		return false, err
 	}
 	if transitGatewayID != nil {
 		s.SetStatus(infrav1beta2.ResourceTypeTransitGateway, infrav1beta2.ResourceReference{ID: transitGatewayID, ControllerCreated: ptr.To(true)})
-		return nil
 	}
-	return nil
+	return true, nil
 }
 
 // checkTransitGateway checks transit gateway exist in cloud.
-func (s *PowerVSClusterScope) checkTransitGateway() (string, error) {
+func (s *PowerVSClusterScope) checkTransitGateway() (string, bool, error) {
 	// TODO(karthik-k-n): Support regex
 	transitGateway, err := s.TransitGatewayClient.GetTransitGatewayByName(*s.GetServiceName(infrav1beta2.ResourceTypeTransitGateway))
 	if err != nil {
-		return "", err
+		return "", false, err
 	}
 	if transitGateway == nil || transitGateway.ID == nil {
-		return "", nil
+		return "", false, nil
 	}
-	if err = s.checkTransitGatewayStatus(transitGateway.ID); err != nil {
-		return "", err
+	requeue, err := s.checkTransitGatewayStatus(transitGateway.ID)
+	if err != nil {
+		return "", false, err
 	}
-	return *transitGateway.ID, nil
+	return *transitGateway.ID, requeue, nil
 }
 
 // checkTransitGatewayStatus checks transit gateway status in cloud.
-func (s *PowerVSClusterScope) checkTransitGatewayStatus(transitGatewayID *string) error {
+func (s *PowerVSClusterScope) checkTransitGatewayStatus(transitGatewayID *string) (bool, error) {
+	requeue := false
 	transitGateway, _, err := s.TransitGatewayClient.GetTransitGateway(&tgapiv1.GetTransitGatewayOptions{
 		ID: transitGatewayID,
 	})
 	if err != nil {
-		return err
+		return requeue, err
 	}
 	if transitGateway == nil {
-		return fmt.Errorf("tranist gateway is nil")
+		return requeue, fmt.Errorf("transit gateway is nil")
 	}
-	if *transitGateway.Status != string(infrav1beta2.TransitGatewayStateAvailable) {
-		return fmt.Errorf("error tranist gateway %s not in available status, current status: %s", *transitGatewayID, *transitGateway.Status)
+
+	switch *transitGateway.Status {
+	case string(infrav1beta2.TransitGatewayStateAvailable):
+		s.Info("Transit gateway successfully created and is in available state")
```

Comment:
since we are checking status and not always creating , this line may mislead user

Reply:
That makes sense, changed it.